openhands/server/session/agent_session.py

```diff
@@ -199,6 +210,12 @@ async def _create_runtime(
             return
 
         if self.runtime is not None:
```

Comment:
We can remove this `not None` check--should never happen

Reply:
Is it never possible for runtime_cls() to return None?

Reply:
correct. And we call `runtime.connect()` above, so if it's None we already have a problem :)